src/metabase/sync/fetch_metadata.clj

```diff
@@ -28,25 +30,56 @@
   (log-if-error "db-metadata"
     (driver/describe-database (driver.u/database->driver database) database)))
 
-(mu/defn fields-metadata
-  "Effectively a wrapper for [[metabase.driver/describe-fields]] that also validates the output against the schema."
-  [database :- i/DatabaseInstance & {:as args}]
-  (log-if-error "fields-metadata"
-    (cond->> (driver/describe-fields (driver.u/database->driver database) database args)
-      ;; This is a workaround for the fact that [[mu/defn]] can't check reducible collections yet
-      (mu.fn/instrument-ns? *ns*)
-      (eduction (map #(mu.fn/validate-output {} i/FieldMetadataEntry %))))))
-
 (mu/defn table-fields-metadata :- [:set i/TableMetadataField]
-  "Get more detailed information about a `table` belonging to `database`. Includes information about the Fields."
+  "Fetch metadata about Fields belonging to a given `table` directly from an external database by calling its driver's
+  implementation of [[driver/describe-table]], or [[driver/describe-fields]] if implemented. Also includes nested field
+  column metadata."
   [database :- i/DatabaseInstance
    table    :- i/TableInstance]
   (log-if-error "table-fields-metadata"
-    (if (driver/database-supports? (driver.u/database->driver database) :describe-fields database)
-      (set (fields-metadata database :table-names [(:name table)] :schema-names [(:schema table)]))
-      (:fields (driver/describe-table (driver.u/database->driver database) database table)))))
+    (let [driver (driver.u/database->driver database)
+          result (if (driver/database-supports? driver :describe-fields database)
+                   (set (driver/describe-fields driver
+                                                database
+                                                :table-names [(:name table)]
+                                                :schema-names [(:schema table)]))
+                   (:fields (driver/describe-table driver database table)))]
+      (cond-> result
+        (driver/database-supports? driver :nested-field-columns database)
+        ;; TODO: decouple nested field columns sync from field sync. This will allow
```

Comment:
Should we make a tech debt issue for this?

Reply:
This is probably better off in a bigger project. I'm writing a tech-driven product doc for it as we speak.

Reply:
The gist of the comment is that non-trivial TODOs should include a link back to an issue, or I guess a doc link could work.